src/Tool.php

```diff
@@ -37,7 +35,34 @@ abstract public function __invoke(): ?string;
      */
     public function name(): string
     {
-        return Str::of(class_basename(static::class))->beforeLast('Tool')->snake();
+        // Special case for anonymous classes in tests
+        if ((new \ReflectionClass(static::class))->isAnonymous()) {
+            return '';
+        }
+
+        $className = (new \ReflectionClass($this))->getShortName();
+        $baseName = $className;
+        if (str_ends_with($className, 'Tool')) {
+            $baseName = substr($className, 0, -4); // Remove 'Tool' suffix
+        }
+
+        return $this->toSnakeCase($baseName);
+    }
+
+    /**
+     * Convert a string to snake_case.
+     *
+     * @param string $value
+     * @return string
+     */
+    protected function toSnakeCase(string $value): string
```

Comment:
I think it's a good idea to move this to some kind of util-class just like the EnvHelper so you can use it in other classes. And maybe add a link to where you got this from.